builtin/logical/pki/path_issue_sign.go

```diff
@@ -219,6 +219,14 @@ func (b *backend) pathSignVerbatim(ctx context.Context, req *logical.Request, da
 		entry.Issuer = role.Issuer
 	}
 
+	// Either we have a non-sign-verbatim path, or we have a role named
+	// sign-verbatim being used with the sign-verbatim endpoint.
+	pathShouldHaveRole := !strings.HasSuffix(req.Path, "sign-verbatim")
```

Comment:
Instead of all this should we not fix this within backend.metricsWrap ? If it was provided a roleName and it failed to lookup the role to fail?

Reply:
Ahhh, that's probably better.